CleverRaven/Cataclysm-DDA
# Added explicit num capture state to drop/clean menus to allow numpad navigation

#### Summary
Bugfixes "Added explicit num capture state to drop/clean menus to allow numpad navigation"


#### Purpose of change
Fixes #50860 and Fixes #50819 


#### Describe the solution
As discussed on #50860, an explicit state was added that captured number keys. This state can only be entered if the new keybinding (MARK_WITH_COUNT) is bound to a key, I recommend '#' or '+', and when it is bound you can use numpad number keys for navigation within the affected menus (mutli drop and clean). If it isn't bound, which it isn't by default, then the menu behavior stays the same where it captures all number keys and it prevents numpad navigation. I've also added a display of the current number for both number entry methods where the filter text shows up. This means the legacy method of selecting uses <enter> instead of TOGGLE_ENTRY to choose the number, I feel this is appropriate now that the user can see the numbers they are typing.



#### Describe alternatives you've considered

As far as macro implementation:
Letting one of the other Drop/Clean menu PRs handle this. Potentially migrated inventory menus to use keycodes.

Micro implementation:
Rather than introducing a new bool variable to track num capture, do something fancy and reuse count as an indicator (setting to -1 or something).



#### Testing

With MARK_WITH_COUNT unbound:

1. Spawn character with multiple of an item
2. Enter drop menu, highlight item with quantity > 4
3. Move selector with arrow keys to select the item and type "4" and verify the count entry area shows a 4
4. Press enter to note that 4 of of the item have been selected for dropping
5. Perform a similar action with the clean menu, summoning 10 filthy toothbrushes, a washing kit, and 20 soap.

With MARK_WITH_COUNT bound to "+":

1. Spawn character with multiple of an item
2. Enter drop menu, highlight item with quantity > 4
3. Move selector with numpad "2" to select the item then press "+", verify count entry shows and type 4, verify it now shows 4.
4. Press enter to note that 4 of of the item have been selected for dropping
5. Perform a similar action with the clean menu, summoning 10 filthy toothbrushes, a washing kit, and 20 soap.



#### Additional context
